packages/rum-recorder/src/domain/segmentCollection.ts

```diff
@@ -189,7 +189,7 @@ export function computeSegmentContext(applicationId: string, session: RumSession
     return undefined
   }
   const viewContext = parentContexts.findView()
-  if (!viewContext?.session.id) {
+  if (!viewContext?.session.id || viewContext.session.id !== session.getId()) {
```

Comment:
```suggestion
  if (!viewContext || viewContext.session.id !== session.getId()) {
```
Could be a bit simpler

Reply:
Since SegmentContext require `session.id` to be `string`, we still need to check that session id is defined to make the compiler happy.

Reply:
Alright. The typechecking is less safe in `assembly` then, since we don't need this check. Probably masked by the `as RumEvent` cast.

Reply:
We could enforce ViewContext session id to be defined, it should not hurt

Reply:
FMU currently a `viewContext` *may*  have an undefined session id, for example if the session expire and the URL changes without any user interaction. It is not an issue because we correctly check if the session is tracked before assembling the event, but it may require a bit more work that expected.

Reply:
ok, it could worth to improve that at some point